src/pocketmine/entity/Entity.php

```diff
@@ -610,6 +610,10 @@ public function __construct(Level $level, CompoundTag $nbt){
 
 		$this->fallDistance = $this->namedtag->getFloat("FallDistance", 0.0);
 
+		if(!isset($this->width) or !isset($this->height)){
+			throw new \InvalidStateException(static::class . " has invalid width/height");
+		}
+
```

Comment:
It is possible to replace it with the following code, but the behavior is difficult to understand and it is not friendly to the reviewer.
```php
if(!isset($this->width, $this->height)){
```

Reply:
Does InvalidStateException need to be changed to another exception ...?

Reply:
With the advice of alvin0319, I changed from InvalidStateException to AssumptionFailedError.

Reply:
`AssumptionFailedError` is to be used in places where we assume the exception will never be thrown (e.g. a builtin function that never returns false with the given inputs). I don't think it fits here because this is a case which can actually occur due to improper code.